Add 2D circle EB test

KZhu-ME opened the pull request:
To detect if any future update breaks the 2D embedded boundary use case.

- [x] rebase after #2401 was merged
- [x] add load balancing to the test case so we can cover #2554

KZhu-ME commented:
Test currently fails due to https://github.com/ECP-WarpX/WarpX/issues/2534, but should be fixed with https://github.com/ECP-WarpX/WarpX/pull/2536.

ax3l commented on Regression/WarpX-tests.ini:
```diff
@@ -2245,6 +2245,24 @@ compareParticles = 0
+dim = 2
+addToCompileString = USE_EB=TRUE
+restartTest = 0
+useMPI = 1
+numprocs = 2
+useOMP = 1
+numthreads = 1
```
Looking good :)

ax3l commented on Examples/Tests/embedded_circle/inputs_2d:
```diff
@@ -0,0 +1,80 @@
+electrons.save_particles_at_zlo = 1
+electrons.save_particles_at_zhi = 1
+electrons.save_particles_at_eb = 1
+ar_ions.mass = 6.633520326333601e-26
+ar_ions.charge = q_e
```
Shall we add argon to the pre-defines particle types? :)
(as a follow-up PR, I mean)

KZhu-ME replied:
That sounds like a good idea! We'll open up another PR for that.

ax3l commented on Examples/Tests/embedded_circle/inputs_2d:
```diff
@@ -0,0 +1,80 @@
+ar_ions.save_particles_at_zlo = 1
+ar_ions.save_particles_at_zhi = 1
+ar_ions.save_particles_at_eb = 1
+collisions.collision_names = coll_electrons coll_ar_ions
```
```suggestion

collisions.collision_names = coll_electrons coll_ar_ions
```

ax3l commented on Examples/Tests/embedded_circle/inputs_2d:
```diff
@@ -0,0 +1,80 @@
+warpx.eb_implicit_function = -((x-0.00005)**2+(z-0.00005)**2-1e-05**2)
+warpx.eb_potential(x,y,z,t) = -10
+warpx.self_fields_absolute_tolerance = 0.02
+amr.n_cell = 128 128
```
Just lightening up the wall of text into logical groups a bit :)
```suggestion

amr.n_cell = 128 128
```

ax3l commented on Examples/Tests/embedded_circle/inputs_2d:
```diff
@@ -0,0 +1,80 @@
+boundary.particle_lo = absorbing periodic
+boundary.particle_hi = absorbing periodic
+algo.particle_shape = 1
+particles.species_names = electrons ar_ions
```
```suggestion

particles.species_names = electrons ar_ions
```

lgiacome commented on Examples/Tests/embedded_circle/inputs_2d:
```diff
@@ -0,0 +1,80 @@
+max_step = 1
```
Since this is a rather complex test and its name is not self-explanatory it would be probably a good idea to explain in a few lines what the test does.

KZhu-ME replied:
Sounds good, I'll add that in.

lgiacome reviewed (changes requested):
That's a very cool and comprehensive test. I just suggested adding a description of the test at the beginning of the input file so that the test can be also instructive for new users as a reference of what the code can do.

ax3l commented:
This is already green, but we know that 2D EB shows undefined behavior until we merge #2401 

Quoting @lgiacome:
> Anytime that `WarpX_EB` is `ON` we call some initialization functions which are broken in 2D and #2401 fixes them. Now, these functions essentially assemble the `m_edge_lengths`  and `m_face_areas`  multifabs, which are not needed for the electrostatic solver for example.
Eventually we should avoid calling these functions when only the ES solver is used.

ax3l commented:
@KZhu-ME ready for rebase/merge of `development` (#2401 now merged :))

KZhu-ME commented:
I have rebased `development` into this branch and added load balancing to this test. However, with the addition of load balancing, the test will fail.

ax3l commented:
Ok sounds good.
We can add the test coverage for now w/o load balancing and one the we fixed the latter, we can ensure this does not break again by adding load balancing in a follow-up to the test as well.

ax3l commented:
Oh: Funny enough, test did not fail (with the two MPI ranks in CI).

KZhu-ME commented:
Oh huh, that's interesting, the test case fails for me locally...

ax3l reviewed (approved):
Thank you! :)
